desktop/src/main/java/bisq/desktop/main/dao/wallet/send/BsqSendView.java

```diff
@@ -282,16 +322,49 @@ private void onSendBsq() {
         }
     }
 
+    private void onBsqInputControl() {
+        List<TransactionOutput> unspentTransactionOutputs = bsqWalletService.getSpendableBsqTransactionOutputs();
+        if (bsqUtxoCandidates == null) {
+            bsqUtxoCandidates = new HashSet<>(unspentTransactionOutputs);
+        } else {
+            // If we had some selection stored we need to update to already spent entries
+            bsqUtxoCandidates = bsqUtxoCandidates.stream().
+                    filter(e -> unspentTransactionOutputs.contains(e)).
+                    collect(Collectors.toSet());
```

Comment:
This would remove transactions that are no longer spendable BSQ outputs from the selected outputs. I think that could be confusing. Better to check if the `bsqUtxoCandidates` are still all spendable, if they are then all is good, if not, then reset the selection to allow choosing between all `unspentTransactionOutputs` again.

Reply:
To request the spendable via bsqWalletService.getSpendableBsqTransactionOutputs() and then check if the bsqUtxoCandidate is in that set does check if the bsqUtxoCandidate is spendable. We use the coinSelector for that and there is no API for checking one utxo if it is still spendable.

Or do I misunderstand what you mean?